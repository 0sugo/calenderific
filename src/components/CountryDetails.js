import React from 'react';
import { useSelector } from 'react-redux';
import Navbar from './Navbar';

const CountryDetails = () => {
  const { country } = useSelector((store) => store.countryDetails);
  // console.log(country);
  return (
    <div>
      <Navbar />
      <div className="chosen-container">
        {country.map((item) => (
          <div className="chosen-holidays" key={item.date}>
            <p>{item.name}</p>
            <p className="dater">{item.date}</p>
          </div>
        ))}
      </div>
    </div>
  );
};
export default CountryDetails;
