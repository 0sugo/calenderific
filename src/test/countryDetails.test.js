import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom'; // Import the MemoryRouter component
import CountryDetails from '../components/CountryDetails';

const mockStore = configureStore([]);

test('renders the CountryDetails component correctly', () => {
  const initialState = {
    countryDetails: {
      country: [
        {
          countryCode: 'US',
          name: 'United States',
          date: '2023-01-01',
        },
        {
          countryCode: 'US',
          name: 'Independence Day',
          date: '2023-07-04',
        },
      ],
    },
    allHolidays: {
      AllHolidays: [
        {
          countryCode: 'US',
          name: 'United States',
        },
        {
          countryCode: 'CA',
          name: 'Canada',
        },
      ],
    },
  };

  const store = mockStore(initialState);

  const { container } = render(
    <Provider store={store}>
      <MemoryRouter> {/* Wrap the component with the MemoryRouter */}
        <CountryDetails />
      </MemoryRouter>
    </Provider>
  );

  expect(container).toMatchSnapshot();
});
