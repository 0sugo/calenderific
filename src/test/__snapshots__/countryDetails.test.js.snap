// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CountryDetails component correctly 1`] = `
<div>
   
  <div>
    <div
      class="nav-container"
    >
      <a
        aria-current="page"
        class="back active"
        href="/"
      >
        <svg
          class="w-6 h-6 leftchev"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 19.5L8.25 12l7.5-7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </a>
      <h1
        class="topper"
      >
        World Holidays
      </h1>
    </div>
    <div
      class="hero-section"
    >
      <img
        alt="world"
        class="header-photo"
        src="globalphoto.png"
      />
      <div
        class="All-holidays"
      >
        <h3>
          <span
            class="before-text"
          >
            United States
          </span>
          <br />
          <span
            class="main-text"
          >
            2
          </span>
           
          <span
            class="after-text"
          >
            Holidays
          </span>
        </h3>
      </div>
    </div>
    <div
      class="chosen-container"
    >
      <h3
        class="Banner"
      >
        HOLIDAYS IN
         
        UNITED STATES
         - 2023
      </h3>
      <div
        class="chosen-holidays"
      >
        <p>
          United States
        </p>
        <p
          class="dater"
        >
          2023-01-01
        </p>
      </div>
      <div
        class="chosen-holidays"
      >
        <p>
          Independence Day
        </p>
        <p
          class="dater"
        >
          2023-07-04
        </p>
      </div>
    </div>
  </div>
</div>
`;
